File: Common/Models/Graphql/Logs/MemoryLog.cs
namespace YALM.Common.Models.Graphql.Logs;

public class MemoryLog : LogBase
{
	public long? TotalKb { get; set; }
	public long? FreeKb { get; set; }
	public long? UsedKb { get; set; }
	public long? SwapTotalKb { get; set; }
	public long? SwapFreeKb { get; set; }
	public long? SwapUsedKb { get; set; }
	public long? AvailableKb { get; set; }
	public long? CachedKb { get; set; }
}

File: website/src/models/MemoryLog.ts
export class MemoryLog{
  date: string;
  totalKb: number;
  usedKb: number;
  freeKb: number;
  availableKb: number;
  cachedKb: number;
  swapTotalKb: number;
  swapUsedKb: number;
  swapFreeKb: number;

  constructor(date: string, totalKb: number, usedKb: number, freeKb: number, availableKb: number, cachedKb: number, swapTotalKb: number, swapUsedKb: number, swapFreeKb: number) {
    this.date = date
    this.totalKb = totalKb
    this.usedKb = usedKb
    this.freeKb = freeKb
    this.availableKb = availableKb
    this.cachedKb = cachedKb
    this.swapTotalKb = swapTotalKb
    this.swapUsedKb = swapUsedKb
    this.swapFreeKb = swapFreeKb
  }

}